src/api/app/controllers/webui/projects/maintained_projects_controller.rb

```diff
@@ -0,0 +1,56 @@
+module Webui
+  module Projects
+    class MaintainedProjectsController < WebuiController
+      before_action :set_project
+      before_action :set_maintained_project, except: [:index]
+      after_action :verify_authorized, except: [:index]
+
+      def index
+        respond_to do |format|
+          format.html
+          format.json do
+            render json: MaintainedProjectDatatable.new(params, view_context: view_context,
+                                                                project: @project, current_user: User.current)
+          end
+        end
+        switch_to_webui2
+      end
+
+      def destroy
+        authorize @project, :destroy?
+
+        maintenance_project = MaintainedProject.find_by!(project: @maintained_project)
+
+        @project.maintained_projects.destroy(maintenance_project)
+
+        flash_message = if @project.store
```

Comment:
Note that `store` calls `save!`. So it might raise an error, e.g. in case of validation issues.
And the `write_to_backend` method [might](https://github.com/openSUSE/open-build-service/blob/master/src/api/lib/backend/connection.rb#L107) raise a `Timeout::Error`.

Reply:
@bgeuken ,

so what is your suggestion? ```@project.valid? && @project.store``` ?

Reply:
That wouldn't help in case that `write_to_backend` raises an exception. I fear you have to catch this.

One way or the other it would be nice to add the error message to the output for the user. At least in case of validation errors, since there we know we can show them to users.
In case of backend timeouts it's probably enough to state that there has been an internal error and the user should try again.

Reply:
We discussed this in private. We will use `@project.valid? && @project.store` here and I am going to write a generic exception handler for the timeouts / backend errors that happen when calling write_to_backend.

Reply:
@bgeuken _Senk ju_ <3 please review it again

Reply:
-> https://github.com/openSUSE/open-build-service/pull/7423

Reply:
@vpereira This won't cover the validation error. So you would still have to check for projects being `*.valid?`.